app/controllers/admin_controller.rb

```diff
@@ -1,5 +1,18 @@
 class AdminController < ApplicationController
+  before_action :ensure_admin
+
   def index
+    redirect_to admin_reports_url if current_user.role == 'admin'
```

Comment:
now /admin route is not accessible as we are redirecting in both cases
create a Admin::BaseController and inherit all controller in admin namespace from it